### ISSUE TITLE
Automation

### ISSUE THREAD

**Issue by kleuter** [crystalidea/macs-fan-control#373]
Automate switching profiles in Macs Fan Control, e.g. a CLI, TCP socket, or Automator etc

**Comment by robbyt** [comment]
Just a bit more info on this, I recently picked up an Elgato Stream Deck, and I would like to program some buttons to switch between fan profiles on my MacPro.

**Comment by kleuter** [comment]
Some more ideas from users:

> It would be great if we could interact with Macs Fan Control externally. The most useful feature to control would be changing the active preset.

Possible methods:

- command line utility that can accept the preset name as an argument
- adding a Scripting Bridge dictionary to the app so that AppleScript can tell Macs Fan Control to do things, like change the active preset
- adding a custom URI handler, such as macsfancontrol://setactivepreset/_PresetName_/

Any of those solutions would be fine, or all of them would be great.

Use case: during the day, I want a more moderate preset because I work in a quiet studio so I want the fans to run slower, but when the screen saver turns on or the computer is left idle overnight, I want a more aggressive fan preset to engage to cool certain components more.

Those system event triggers can be handled in a 3rd party macro app like Keyboard Maestro, but it needs a way to communicate with Macs Fan Control to tell it to change a preset.

**Comment by zecanard** [comment]
AppleScript would be amazing. Thanks!

**Comment by ApopheniaPays** [comment]
Upvote, I could really use some sort of scripted control.

**Comment by zecanard** [comment]
#451 could also be implemented by scripting support, if not built-in.

**Comment by jeffbargmann** [comment]
Given the audience MFC targets, this would also seem to provide some affordance as well to folks who want keyboard shortcuts. https://github.com/crystalidea/macs-fan-control/issues/348 (Having "both" features is nice yeah, but in mean time, most of us wouldn't have a issue wiring up Alfred/etc.)

**Comment by jeremychone** [comment]
Yes, a command line interface to set the preset would be greatly appreciated. Right now, I did an Automator script that clicks on the app menu, but cumbersome and distracting. A command line `--set-preset full` would be great.

**Comment by kibin** [comment]
+1 for this feature
also it can be implemented as shortcuts integration maybe, rogue amoeba's soundsource did it some time ago and it works nicely with hammerspoon and automators alike

**Comment by ansh-les** [comment]
+1 for scripting control

**Comment by Caffeine19** [comment]
+1, support for a URL scheme would also be great. I'd like to switch presets using Raycast.
